tests/end_to_end_test/snapshot_create_test.go

```diff
@@ -471,6 +472,79 @@ func TestSnapshotCreateAllWithManualSnapshot(t *testing.T) {
 	e.RunAndVerifyOutputLineCount(t, expectedSnapshotCount, "snapshot", "list", "--show-identical", "-a")
 }
 
+func TestSnapshotCreateWithStdinStream(t *testing.T) {
+	t.Parallel()
+
+	e := testenv.NewCLITest(t)
+
+	defer e.RunAndExpectSuccess(t, "repo", "disconnect")
+	e.RunAndExpectSuccess(t, "repo", "create", "filesystem", "--path", e.RepoDir)
+
+	// Create a temporary pipe file with test data
+	content := []byte("Streaming Temporary file content")
+
+	r, w, err := os.Pipe()
+	if err != nil {
+		t.Fatalf("error creating pipe file: %v", err)
+	}
+
+	if _, err = w.Write(content); err != nil {
+		t.Fatalf("error writing to pipe file: %v", err)
+	}
+
+	w.Close()
+
+	// Use the pipe file as stdin
+	initialStdin := os.Stdin
```

Comment:
this modifies global state and will potentially mess up other tests. 

Instead you can pass stdin Reader to subprocess only (exec.Cmd.Stdin), without having to change current process.